backend/src/packages/orders/order.service.ts

```diff
@@ -0,0 +1,168 @@
+  public constructor(orderRepository: OrderRepository) {
+    this.orderRepository = orderRepository;
+  }
+
+  public async create(
+    payload: OrderCreateRequestDto & {
+      userId: number | null;
+      businessId: number;
+    },
+  ): Promise<OrderCreateResponseDto> {
+    const {
+      scheduledTime,
+      carsQty,
+      startPoint,
+      endPoint,
+      customerName,
+      customerPhone,
+      driverId,
+      userId,
+      businessId,
+    } = payload;
+    const order = await this.orderRepository.create(
+      OrderEntity.initializeNew({
+        price: 100, //Mock, get price from driver->user->truck and calculated distance
+        scheduledTime,
+        carsQty,
+        startPoint,
+        endPoint,
+        status: OrderStatus.PENDING,
+        userId,
+        businessId,
+        driverId,
+        customerName,
+        customerPhone,
+      }),
+    );
+
+    return order.toObject();
+  }
+
+  public async findById(
+    id: OrderEntityT['id'],
+    user: UserEntityObjectWithGroupT | null | undefined,
+  ): Promise<OrderEntityT | null> {
+    const foundOrder = await this.orderRepository.findById(id);
+
+    if (!foundOrder) {
+      throw new OrderNotFound();
+    }
+    const result = foundOrder.toObject();
+
+    if (user) {
+      this.checkIfOrderBelongsToUser(result, user);
+    }
+
+    return result;
+  }
+
+  public async update(parameters: {
+    id: OrderEntityT['id'];
+    payload: Partial<OrderEntityT>;
+    driverId: number | undefined;
+  }): Promise<OrderUpdateResponseDto> {
+    const foundOrder = await this.orderRepository.findById(parameters.id);
+
+    if (parameters.driverId && foundOrder) {
+      this.checkIfOrderBelongsToDriver(
+        foundOrder.toObject(),
+        parameters.driverId,
+      );
+    }
+
+    const updatedOrder = await this.orderRepository.update(parameters);
+
+    if (!updatedOrder) {
+      throw new OrderNotFound();
+    }
+
+    return updatedOrder.toObject();
+  }
+
+  public async find({
+    userId,
+    businessId,
+    driverId,
+    currentUserBusinessId,
+    currentUserId,
+  }: {
+    userId: string | undefined;
+    businessId: string | undefined;
+    driverId: string | undefined;
+    currentUserBusinessId: number | undefined;
+    currentUserId: number | null;
+  }): Promise<{ items: OrderEntityT[] }> {
+    const usersOrders = await this.orderRepository.find({
+      userId: Number(currentUserBusinessId ? userId : currentUserId),
+      businessId: Number(currentUserBusinessId ?? businessId),
+      driverId: Number(driverId),
+    });
+
+    return {
+      items: usersOrders.map((it) => it.toObject()),
+    };
+  }
+
+  public async delete(
+    id: OrderEntityT['id'],
+    user: UserEntityObjectWithGroupT,
+  ): Promise<boolean> {
+    const foundOrder = await this.findById(id, user);
+
+    if (!foundOrder) {
+      throw new OrderNotFound();
+    }
+
+    return await this.orderRepository.delete(id);
+  }
+
+  private checkIfOrderBelongsToDriver(
```

Comment:
methods that are called with check are usually retuning boolean value. So, you should better rename method, or return boolean value from this method

Reply:
fixed